dcpy/utils/data.py

```diff
@@ -118,6 +119,17 @@ def read_data_to_df(
                         geometry=gpd.points_from_xy(df[x_column], df[y_column]),
                         crs=csv.geometry.crs,
                     )
+        case file.Json():
+            with open(local_data_path) as f:
+                json_str = json.load(f)
+            if data_format.normalize:
+                gdf = pd.json_normalize(json_str, **data_format.normalize)
+            else:
+                gdf = pd.read_json(json_str)
+            # Feels a tad hacky to have this here instead of in preprocessing
+            # But json columns were causing parquet issues
+            if data_format.columns:
+                gdf = gdf[data_format.columns]
```

Comment:
curious what the parquet issues are

does this mean that if a template doesn't have a `columns` list then `read_data_to_df` might cause parquet issues?

Reply:
This was specifically conversion of a json column - and seems to be a pandas/pyarrow serialization issue. Hence why this is only defined in the json source format. I only needed it for this specific dataset